sqld/src/http/db_factory.rs

```diff
@@ -0,0 +1,64 @@
+use std::sync::Arc;
+
+use axum::extract::FromRequestParts;
+use bytes::Bytes;
+use hyper::http::request::Parts;
+use hyper::HeaderMap;
+
+use crate::connection::MakeConnection;
+use crate::database::Database;
+use crate::error::Error;
+use crate::namespace::MakeNamespace;
+use crate::DEFAULT_NAMESPACE_NAME;
+
+use super::AppState;
+
+pub struct MakeConnectionExtractor<D>(pub Arc<dyn MakeConnection<Connection = D>>);
+
+#[async_trait::async_trait]
+impl<F> FromRequestParts<AppState<F>>
+    for MakeConnectionExtractor<<F::Database as Database>::Connection>
+where
+    F: MakeNamespace,
+{
+    type Rejection = Error;
+
+    async fn from_request_parts(
+        parts: &mut Parts,
+        state: &AppState<F>,
+    ) -> Result<Self, Self::Rejection> {
+        let ns = namespace_from_headers(&parts.headers, state.allow_default_namespace)?;
+        Ok(Self(
+            state
+                .namespaces
+                .with(ns, |ns| ns.db.connection_maker())
+                .await?,
+        ))
+    }
+}
+
+pub fn namespace_from_headers(
```

Comment:
This is actually not an issue since we pass namespace via protobuf message and that is where this issue with domains is. So this is no longer a problem.